src/test/software/amazon/event/ruler/RuleCompilerTest.java

```diff
@@ -220,7 +227,8 @@ public void testFlattenRule() throws Exception {
                 "\"c1\": [" +
                 "{ \"suffix\": \"child\" }," +
                 "{ \"anything-but\": [111,222,333]}," +
-                "{ \"anything-but\": { \"prefix\": \"foo\"}}" +
+                "{ \"anything-but\": { \"prefix\": \"foo\"}}," +
+                "{ \"anything-but\": { \"suffix\": \"oof\"}}" +
```

Comment:
nit: this initially made me wonder if we need reverse of `foo` for suffix matching as a condition (github's review pane was hiding the function names). Using a different work would any prevent any such assumptions from ever occurring.

Reply:
changed